test/unit/org/apache/cassandra/db/marshal/ValueAccessorTest.java

```diff
@@ -39,6 +39,28 @@
 
 public class ValueAccessorTest
 {
+    /**
+     * This method should be used to test values to be processed by {@link ValueAccessor}
+     * in order to make sure we do not assume position == 0 in the underlying {@link ByteBuffer}
+     */
+    public static <V> V maybeCopyWithPadding(V value)
```

Comment:
Why not call it `maybePad`? The important bit here is whether it is being padded or not., not that is being copied or not. So when I first read it I was wondering why being copied or not was a thing.

Also why pad in a 'random' way and not with a fix value? You can get 0 padding which beats the purpose of the test imo and if it failed you wouldn't know what was the padding value.

Reply:
I'll rename the method and keep some kind of randomness. I'd like to avoid the situation that the subsequent paddings are of the same length.

Reply:
Sgtm. What about printing in stdout the used padding or alternatively including the padded values in the assert msg? Sorry I usually do a lot of test fixing so I tend to ocd a bit on these test things that help me a lot lol

Reply:
The those tests are already relying heavily on randomness. Having several level of randomness sound like a bad idea to me unless we do it proprely. Logging some extra data along the way is far from a perfect solution. The `testUnsignedShort` test already relies on `quicktheories` which produce a really clear output on failure allowing somebody to easily reproduce the problem. It is possible to use that library to add the extra randomness you need for testing your change. Doing it for the others tests will require more work as they do not relies on `quicktheories`.     

We all have suffered a lot trying to get ride of flaky tests and the experience as showed us that unless people are easily able to reproduce the problem they will chose to wait for somebody else to fix it.

Reply:
ok, i didn't know quicktheories, please have a look at my recent changes